src/blk/BlockDevice.h

```diff
@@ -189,6 +188,7 @@ class BlockDevice {
   uint64_t zone_size = 0;
 
 public:
+  bool support_discard = false;
```

Comment:
let's have a getter func (e.g. bool is_discard_supported()) to check this var's value instead of making the variable explicitly public.